// @aws-amplify/core/lib/clients/index.js
"use strict";
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseMetadata = exports.parseJsonError = exports.parseJsonBody = exports.presignUrl = exports.signRequest = exports.composeServiceApi = exports.authenticatedHandler = exports.unauthenticatedHandler = exports.fetchTransferHandler = exports.getDnsSuffix = void 0;
var endpoints_1 = require("./endpoints");
Object.defineProperty(exports, "getDnsSuffix", { enumerable: true, get: function () { return endpoints_1.getDnsSuffix; } });
var fetch_1 = require("./handlers/fetch");
Object.defineProperty(exports, "fetchTransferHandler", { enumerable: true, get: function () { return fetch_1.fetchTransferHandler; } });
var unauthenticated_1 = require("./handlers/unauthenticated");
Object.defineProperty(exports, "unauthenticatedHandler", { enumerable: true, get: function () { return unauthenticated_1.unauthenticatedHandler; } });
var authenticated_1 = require("./handlers/authenticated");
Object.defineProperty(exports, "authenticatedHandler", { enumerable: true, get: function () { return authenticated_1.authenticatedHandler; } });
var composeServiceApi_1 = require("./internal/composeServiceApi");
Object.defineProperty(exports, "composeServiceApi", { enumerable: true, get: function () { return composeServiceApi_1.composeServiceApi; } });
var signatureV4_1 = require("./middleware/signing/signer/signatureV4");
Object.defineProperty(exports, "signRequest", { enumerable: true, get: function () { return signatureV4_1.signRequest; } });
Object.defineProperty(exports, "presignUrl", { enumerable: true, get: function () { return signatureV4_1.presignUrl; } });
var serde_1 = require("./serde");
Object.defineProperty(exports, "parseJsonBody", { enumerable: true, get: function () { return serde_1.parseJsonBody; } });
Object.defineProperty(exports, "parseJsonError", { enumerable: true, get: function () { return serde_1.parseJsonError; } });
Object.defineProperty(exports, "parseMetadata", { enumerable: true, get: function () { return serde_1.parseMetadata; } });
__exportStar(require("./types"), exports);
